staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -125,6 +127,10 @@ func (s *ServerRunOptions) Validate() []error {
 		errors = append(errors, fmt.Errorf("--request-timeout can not be negative value"))
 	}
 
+	if s.GoawayChance < 0 || s.GoawayChance > 1 {
+		errors = append(errors, fmt.Errorf("--goaway-chance can not be less than 0 and greater than 1"))
```

Comment:
It really needs to be quite small to not break the cluster. Personally I don't think I'd let it be larger than 1/500; I'm OK with validation permitting values that aren't a good idea, but I can't imagine running with this over 1/50 would be a good idea. Certainly values approaching 1 are really not good and so 1 is definitely too high..

Reply:
has add validation to limit max value less than 0.02